r/R/dataset-partition.R

```diff
@@ -72,19 +77,22 @@ HivePartitioning$create <- dataset___HivePartitioning
 #' Because fields are named in the path segments, order of fields passed to
 #' `hive_partition()` does not matter.
 #' @param ... named list of [data types][data-type], passed to [schema()]
+#' @param null_fallback character to be used in place of `NA` and `NULL` values
```

Comment:
Self nit: In this comment I say "`NA` and `NULL`" and in the next comment I say `null`.  I guess I wasn't being too consistent.  I'm not sure what the proper R term for "a null array value" is.

Reply:
It's a bit complicated, in R most columns would have `NA`s for missing values (though it's possible that `NULL` would show up in some types, it's not particularly common). I added `NA` in here for the documentation on writing since that's most likely to be what's in the columns. I think it's just fine to have `NULL` above when talking about the hive partitioning. In (almost?) all cases those `NULL`s in the hive partitioning will get translated into `NA`s if/when the data is pulled into R.